src/web/app/components/comment-box/comment-row/comment-row.component.spec.ts

```diff
@@ -45,4 +46,114 @@ describe('CommentRowComponent', () => {
+          commentGiver: 'mockCommentGiver',
+          lastEditorEmail: 'mockEditor@example.com',
+          feedbackResponseCommentId: 12345,
+          commentText: 'Mock comment text',
+          showCommentTo: [],
+          createdAt: new Date().getTime(),
+          lastEditedAt: new Date().getTime(),
+          showGiverNameTo: [],
+        },
+        commentEditFormModel: {
+          commentText: 'Mock comment text for form',
+          isUsingCustomVisibilities: false,
+          showCommentTo: [CommentVisibilityType.GIVER],
+          showGiverNameTo: [CommentVisibilityType.INSTRUCTORS],
+        },
+        isEditing: true,
+      };
+
+      const spy = jest.spyOn(component.visibilityStateMachine, 'allowAllApplicableTypesToSee');
+      component.ngOnChanges();
+      expect(spy).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('triggerCloseEditing', () => {
+    it('should emit closeEditing event', () => {
+        component.closeEditingEvent.emit = jest.fn();
+        component.triggerCloseEditing();
+        expect(component.closeEditingEvent.emit).toHaveBeenCalled();
+    });
+  });
+
+  describe('triggerSaveCommentEvent', () => {
+    it('should emit saveComment event', () => {
+        const spy = jest.spyOn(component.saveCommentEvent, 'emit');
+        component.triggerSaveCommentEvent();
+        expect(spy).toHaveBeenCalled();
+    });
+  });
+
+  describe('triggerDeleteCommentEvent', () => {
+    it('should emit deleteComment event after modal confirmation', async () => {
+      const mockModalRef = { result: Promise.resolve(true) };
+      jest.spyOn((component as any).simpleModalService, 'openConfirmationModal').mockReturnValue(mockModalRef);
+      const emitSpy = jest.spyOn(component.deleteCommentEvent, 'emit');
+      component.triggerDeleteCommentEvent();
+      await mockModalRef.result;
+      expect(emitSpy).toHaveBeenCalled();
+    });
+
+  it('should set visibility settings when originalComment is defined and not following feedback question', () => {
+    component.model = {
+      originalComment: {
+        isVisibilityFollowingFeedbackQuestion: false,
+        commentGiver: 'mockCommentGiver',
+        lastEditorEmail: 'mockEditor@example.com',
+        feedbackResponseCommentId: 12345,
+        commentText: 'Mock comment text',
+        showCommentTo: [CommentVisibilityType.GIVER, CommentVisibilityType.INSTRUCTORS],
+        createdAt: new Date().getTime(),
+        lastEditedAt: new Date().getTime(),
+        showGiverNameTo: [],
+      },
+      commentEditFormModel: {
+        commentText: 'Mock comment text for form',
+        isUsingCustomVisibilities: false,
+        showCommentTo: [CommentVisibilityType.GIVER],
+        showGiverNameTo: [CommentVisibilityType.INSTRUCTORS],
+      },
+      isEditing: true,
+    };
+    const spy = jest.spyOn(component.visibilityStateMachine, 'applyVisibilitySettings');
+    component.ngOnChanges();
+    expect(spy).not.toHaveBeenCalled();
+  });
```

Comment:
another example of a place where we should expect the visibilityStateMachine to be called with a parameter

```javascript
    expect(spyVisibilityStateMachine.applyVisibilitySettings).toHaveBeenCalledWith({
      SHOW_COMMENT: [CommentVisibilityType.GIVER, CommentVisibilityType.INSTRUCTORS],
      SHOW_GIVER_NAME: [CommentVisibilityType.GIVER, CommentVisibilityType.INSTRUCTORS]
    });
```